examples/siro_sandbox/gui_navigation_helper.py

```diff
@@ -167,13 +174,28 @@ def _get_humanoid_walk_hints(
                 if visualize_path:
                     self._viz_path(path)
             else:
-                walk_dir = None
+                # TODO: potentially check for previous rotation
+                distance_multiplier = 1.0
+                # walk_dir = None
 
         if walk_dir is None and target_rot_quat is not None:
             walk_dir = self._compute_forward_dir(target_rot_quat)
             distance_multiplier = 0.0
 
-        return walk_dir, distance_multiplier
+        if (
+            found_path
+            and len(path.points) >= 2
+            and path.geodesic_distance >= dist_to_always_move_forward
+        ):
+            target_rot_quat = None
+
+        if target_rot_quat is not None:
+            # Get the forward direction
+            forward_gaze = target_rot_quat.transform_vector(
```

Comment:
This looks almost identical to the code in `_compute_forward_dir`, and I wonder if the result `forward_gaze` is identical to `forward_dir` elsewhere in this function. Can you refactor to avoid code duplication?